src/main/java/org/aksw/simba/squirrel/frontier/Frontier.java

```diff
@@ -69,6 +69,14 @@
      */
     int getNumberOfPendingUris();
 
+    /**
+     * Add hash values for this given uris. Hash values are contained in the uris.
+     *
+     * @param uris   The given uris.
+     */
+    void addHashValuesForUris(List<CrawleableUri> uris);
```

Comment:
I am not sure whether the front end should be involved in this. It might be even better to try to move all this hash value stuff into the deduplicator (if possible)